src/components/MAPI/organizations/AccessControl/index.tsx

```diff
@@ -0,0 +1,178 @@
+import { Box } from 'grommet';
+import produce from 'immer';
+import { useHttpClient } from 'lib/hooks/useHttpClient';
+import { GenericResponse } from 'model/clients/GenericResponse';
+import PropTypes from 'prop-types';
+import React, { useLayoutEffect, useMemo, useState } from 'react';
+import { useSelector } from 'react-redux';
+import DocumentTitle from 'shared/DocumentTitle';
+import { getLoggedInUser } from 'stores/main/selectors';
+import useSWR from 'swr';
+import AccessControlRoleDescription from 'UI/Display/MAPI/AccessControl/AccessControlDescription';
+import AccessControlRoleDetail from 'UI/Display/MAPI/AccessControl/AccessControlRoleDetail';
+import AccessControlRoleList from 'UI/Display/MAPI/AccessControl/AccessControlRoleList';
+import * as ui from 'UI/Display/MAPI/AccessControl/types';
+
+import {
+  appendSubjectsToRoleItem,
+  createRoleBindingWithSubjects,
+  deleteSubjectFromRole,
+  extractErrorMessage,
+  getOrgNamespaceFromOrgName,
+  getRoleItems,
+  getRoleItemsKey,
+} from './utils';
+
+interface IAccessControlProps
+  extends React.ComponentPropsWithoutRef<typeof Box> {
+  organizationName: string;
+}
+
+const AccessControl: React.FC<IAccessControlProps> = ({
+  organizationName,
+  ...props
+}) => {
+  const orgNamespace = getOrgNamespaceFromOrgName(organizationName);
+
+  const client = useHttpClient();
+  const user = useSelector(getLoggedInUser);
+  const { data, mutate, error } = useSWR<
+    ui.IAccessControlRoleItem[],
+    GenericResponse
+  >(
+    getRoleItemsKey(user, orgNamespace),
+    getRoleItems(client, user!, orgNamespace)
+  );
+
+  const [activeRoleName, setActiveRoleName] = useState('');
+  const activeRole = useMemo(
+    () => data?.find((role) => role.name === activeRoleName),
+    [data, activeRoleName]
+  );
+
+  useLayoutEffect(() => {
```

Comment:
Using `useLayoutEffect` instead of `useEffect` gets rid of displaying the search animation for a split second in this scenario:
- Navigate to the `Access control` tab in Organization details to view a role's details
- Switch to the `General` tab in Organization details
- Switch back to the `Access control` tab
- See the loader for a split second